.github/workflows/unit-tests.yml

```diff
@@ -157,19 +157,25 @@ jobs:
     - name: Build Pandas
       id: build
       uses: ./.github/actions/build_pandas
+      env:
+        CFLAGS: "$CFLAGS -fno-sanitize-recover=all"
```

Comment:
Meson has the config option for `b_sanitize=address,undefined` but I don't think it set this. UBSAN has non-fatal errors, so without this things just get printed to stderr and pytest still continues.

Looks like NumPy does something similar with `halt_on_error=1`, but that didn't seem to stop pytest from continuing as I tried this locally

https://github.com/numpy/numpy/pull/24208/files